libs/ha/ha_common_libs_k8s.py

```diff
@@ -1442,47 +1444,67 @@ def get_node_resource_ids(node_obj, r_type, n_type=None, node_id=None):
         return resp
 
     def delete_single_pod_setting_replica_0(self, master_node_obj, health_obj,
-                                            pod_prefix=common_const.POD_NAME_PREFIX):
+                                            pod_prefix=common_const.POD_NAME_PREFIX,
+                                            down_method=common_const.RESTORE_SCALE_REPLICAS):
         """
         Delete Single pod by setting replica set to 0. Check service status is in degraded mode.
         :param master_node_obj: Master node object list
         :param health_obj: Health object
         :param pod_prefix: Pod prefix to be deleted.
+        :param down_method: Pod shutdown/delete method.
         return : tuple
         """
+
+        down_info = {"method": None, "deployment_name": None, "deployment_backup": None}
+        remaining = list()
         LOGGER.info("Get pod name to be deleted")
         pod_list = master_node_obj.get_all_pods(pod_prefix=pod_prefix)
         pod_name = random.sample(pod_list, 1)[0]
```

Comment:
Please consider for multiple pods as well.
Get kvalue as user input.
We can use this method for deleting k pods at once.

Reply:
Yes agree to this, @ankita-93 lets get one single method created for single/multi and 2 diff methods. Once created, whatever chnages needed can be done for @priyankaborawake 's code as well.

Reply:
Done